hicommonwealth/commonwealth
# Migrated CommunitiesController to RQ

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/4975

## Description of Changes
Migrated CommunitiesController to RQ, it only had 1 mutation with help app-level state.

## "How We Fixed It"
By using react query for CommunitiesController mutations

## Test Plan
- Login and join some communities
- Open the explore menu, you will see a list of communities you have joined
- Click on the star icon next to a community to `star` it. Verify it gets starred i.e the star icon fills with yellow color
- Click on the same star icon again. Verify the community gets unstarred i.e the star icon only has a yellow border.

## Deployment Plan
N/A

## Other Considerations
Currently, the star/unstar community is happening by the `/starCommunity` API. I think its a good time to migrate it to be restful. @jnaviask for thoughts.
